lib/rack/session/cookie.rb

```diff
@@ -48,11 +48,17 @@ module Session
     class Cookie < Abstract::Persisted
       # Encode session cookies as Base64
       class Base64
+        def self.strict_encode(str)
+          [str].pack('m0')
+        end
+
         def encode(str)
-          [str].pack('m')
+          Base64.strict_encode(str)
         end
 
         def decode(str)
+          str.unpack('m0').first
+        rescue ArgumentError
```

Comment:
At least on Ruby 2.4.1, trying to `unpack('m0')` with a string that was encoded with `pack('m')` is raising an `ArgumentError` for me:

```
$ ruby -e "p ['A' * 256].pack('m').unpack('m0').first"
-e:1:in `unpack': invalid base64 (ArgumentError)
	from -e:1:in `<main>'
```